File: 11_Container_with_most_water.java
class Solution {
    public int maxArea(int[] height) {
        int l=height.length,area=0,max=0,lptr=0,rptr=l-1;
        for(int x=0;x<l;x++)
        {
            area=(rptr-lptr)*Math.min(height[lptr],height[rptr]);
            max=Math.max(area,max);
            if(height[lptr]>height[rptr])
                rptr-=1;
            else
                lptr+=1;
        }
        return max;
    }
}

File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxArea = float(-inf)
        l,r = 0,len(height) - 1
        while(l < r):
            area = (r-l) * min(height[l],height[r])
            maxArea = max(area,maxArea)
            if(height[l] > height[r]):
                r -= 1
            elif(height[l] <= height[r]):
                l += 1
                
        return maxArea